backend/src/main/java/com/redhat/cloud/notifications/db/repositories/TemplateRepository.java

```diff
@@ -142,7 +143,29 @@ public InstantEmailTemplate findInstantEmailTemplateByEventType(UUID eventTypeId
 
             return instantEmailTemplate;
         } catch (NoResultException e) {
-            throw new NotFoundException("Instant email template not found");
+            throw new NotFoundException(String.format("Event type '%s' doesn't support instant email subscription", eventTypeId));
+        }
+    }
+
+    public AggregationEmailTemplate findAggregationEmailTemplatesByEventType(UUID eventTypeId) {
```

Comment:
The returned object is never used. This method could return a boolean and `FETCH` could be avoided to lower the pressure on the DB. Unless you have a plan to use the method from elsewhere and use the returned data in the near future.

Reply:
sure, I refactored those methods to not fetch results anymore